avr/cores/megacommand/MCL/Wav.cpp

```diff
@@ -270,6 +270,9 @@ bool Wav::read_samples(void *data, uint32_t num_samples, uint32_t sample_index,
     return read_data(data, read_size, position + data_offset);
   }
 
+  if (num_of_channels > 1) {
```

Comment:
`if (num_of_channels == header.fmt.numChannels)`, and can be merged with the above `header.fmt.numChannels == 1`

Reply:
hmm. maybe we do.
if  (channel = 255) then read all channels. 

?

Reply:
honestly I think we only have two practical cases, mono files and stereo files.

Reply:
what about 5.1 surround ? 😉

Reply:
no need for an API to extract channel 1 and 3 from a quadraphonic file.

Reply:
> what about 5.1 surround ? wink

Yeah, I'm wondering what's your intention for loading multi-channel file. Wave editing?

Reply:
yes. wav editor. will probably have a stereo mode. need to find peaks in both channels. 
Much faster to read the stream directly than to buffer each channel.